linkerd/identity/src/lib.rs

```diff
@@ -18,59 +24,56 @@ pub use linkerd_dns_name::InvalidName;
 #[derive(Clone, Debug)]
 pub struct Csr(Arc<Vec<u8>>);
 
-/// An endpoint's identity.
-#[derive(Clone, Eq, PartialEq, Hash)]
-pub struct Name(Arc<linkerd_dns_name::Name>);
+/// An error returned from the TLS implementation.
+pub struct Error(imp::Error);
 
-#[derive(Clone, Debug)]
-pub struct Key(Arc<EcdsaKeyPair>);
+impl stdErr::Error for Error {
+    fn source(&self) -> Option<&(dyn stdErr::Error + 'static)> {
+        stdErr::Error::source(&self.0)
+    }
+}
 
-struct SigningKey(Arc<EcdsaKeyPair>);
-struct Signer(Arc<EcdsaKeyPair>);
+impl fmt::Display for Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.0, fmt)
+    }
+}
 
-#[derive(Clone)]
-pub struct TrustAnchors(Arc<rustls::ClientConfig>);
+impl fmt::Debug for Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Debug::fmt(&self.0, fmt)
+    }
+}
 
-#[derive(Clone, Debug)]
-pub struct TokenSource(Arc<String>);
+impl From<imp::Error> for Error {
+    fn from(err: imp::Error) -> Error {
+        Error(err)
+    }
+}
 
 #[derive(Clone, Debug)]
-pub struct Crt {
-    id: LocalId,
-    expiry: SystemTime,
-    chain: Vec<rustls::Certificate>,
-}
+pub struct Key(imp::Key);
 
-#[derive(Clone)]
-pub struct CrtKey {
-    id: LocalId,
-    expiry: SystemTime,
-    client_config: Arc<rustls::ClientConfig>,
-    server_config: Arc<rustls::ServerConfig>,
-}
+#[derive(Clone, Debug)]
+pub struct TokenSource(Arc<String>);
 
-struct CertResolver(rustls::sign::CertifiedKey);
+#[derive(Clone, Debug)]
+pub struct Crt(imp::Crt);
 
 #[derive(Clone, Debug, Error)]
 #[error(transparent)]
-pub struct InvalidCrt(rustls::TLSError);
+pub struct InvalidCrt(imp::InvalidCrt);
+
+impl From<imp::InvalidCrt> for InvalidCrt {
+    fn from(err: imp::InvalidCrt) -> Self {
+        InvalidCrt(err)
+    }
+}
```

Comment:
nit: the `From` impl could be derived by `thiserror`:
```suggestion
pub struct InvalidCrt(#[from] imp::InvalidCrt);

impl From<imp::InvalidCrt> for InvalidCrt {
    fn from(err: imp::InvalidCrt) -> Self {
        InvalidCrt(err)
    }
}
```